src/main/java/com/mentoring/ecommerce/adapter/in/web/request/ProductReq.java

```diff
@@ -0,0 +1,26 @@
+package com.mentoring.ecommerce.adapter.in.web.request;
+
+import com.sun.istack.internal.NotNull;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.math.BigDecimal;
+
+@Data
+@NoArgsConstructor
+public class ProductReq {
+
+    private String id;
```

Comment:
To save a product it makes no sense to receive an ID. This should be autogenerated by the repository. Please, remove this field from the request class.